examples/add/train_add_distill_sd_wds.py

```diff
@@ -0,0 +1,1762 @@
+
+                                logger.info(
+                                    f"{len(checkpoints)} checkpoints already exist, removing {len(removing_checkpoints)} checkpoints"
+                                )
+                                logger.info(f"removing checkpoints: {', '.join(removing_checkpoints)}")
+
+                                for removing_checkpoint in removing_checkpoints:
+                                    removing_checkpoint = os.path.join(args.output_dir, removing_checkpoint)
+                                    shutil.rmtree(removing_checkpoint)
+
+                        save_path = os.path.join(args.output_dir, f"checkpoint-{global_step}")
+                        accelerator.save_state(save_path)
+                        logger.info(f"Saved state to {save_path}")
+
+                    if global_step % args.validation_steps == 0:
+                        if args.use_ema:
+                            # Store the student unet weights and load the EMA weights.
+                            ema_unet.store(unet.parameters())
+                            ema_unet.copy_to(unet.parameters())
+
+                            log_validation(vae, ema_unet, args, accelerator, weight_dtype, global_step, "ema_student")
+
+                            # Restore student unet weights
+                            ema_unet.restore(unet.parameters())
+
+                        log_validation(vae, unet, args, accelerator, weight_dtype, global_step, "student")
+
+            logs = {
+                "d_total_loss": d_total_loss.detach().item(),
+                "g_total_loss": g_total_loss.detach().item(),
+                "g_adv_loss": g_adv_loss.detach().item(),
+                "g_distill_loss": g_distillation_loss.detach().item(),
+                "lr": lr_scheduler.get_last_lr()[0],
+            }
+            progress_bar.set_postfix(**logs)
+
+            # Write out additional values for accelerator to report.
+            logs["d_adv_loss_fake"] = d_adv_loss_fake.detach().item()
+            logs["d_adv_loss_real"] = d_adv_loss_real.detach().item()
+            logs["d_r1_regularizer"] = d_r1_regularizer.detach().item()
+            logs["d_loss_real"] = d_loss_real.detach().item()
+            accelerator.log(logs, step=global_step)
+
+            if global_step >= args.max_train_steps:
+                break
+
+    # Create the pipeline using using the trained modules and save it.
+    accelerator.wait_for_everyone()
+    if accelerator.is_main_process:
+        unet = accelerator.unwrap_model(unet)
+        unet.save_pretrained(os.path.join(args.output_dir, "unet"))
+
+        # If using EMA, save EMA weights as well.
+        if args.use_ema:
+            ema_unet.store(unet.parameters())
+            ema_unet.copy_to(unet.parameters())
+
+            unet.save_pretrained(os.path.join(args.output_dir, "ema_unet"))
+
+            ema_unet.restore(unet.parameters())
```

Comment:
not necessary here.

Reply:
Removed final `ema_unet.restore(unet.parameters())` call.